packages/store-sync/src/world/getSystemAbi.ts

```diff
@@ -0,0 +1,37 @@
+import { Client, Abi, Address, hexToString, parseAbiItem, Hex } from "viem";
+import metadataConfig from "@latticexyz/world-module-metadata/mud.config";
+import { getRecords } from "../getRecords";
+
+export async function getSystemAbi({
+  client,
+  worldAddress,
+  systemId,
+  fromBlock,
+  toBlock,
+  indexerUrl,
+  chainId,
+}: {
+  readonly client: Client;
+  readonly worldAddress: Address;
+  readonly systemId: Hex;
+  readonly fromBlock?: bigint;
+  readonly toBlock?: bigint;
+  readonly indexerUrl?: string;
+  readonly chainId?: number;
+}): Promise<Abi> {
+  const { records } = await getRecords({
+    table: metadataConfig.tables.metadata__ResourceTag,
+    worldAddress,
+    chainId,
+    indexerUrl,
+    client,
+    fromBlock,
+    toBlock,
+  });
+
+  const abi = records
+    .filter(({ resource, tag }) => hexToString(tag).replace(/\0+$/, "") === "abi" && resource === systemId)
```

Comment:
we have the whole key here now, so we don't need to fetch all records 

```ts
getRecord({
  table: metadataConfig.tables.metadata__ResourceTag,
  ...
  key: { resource: systemId, tag: stringToHexH("abi", { size: 32 }),
});
```

Reply:
yeah makes sense, although `getRecord()` does not exist in `store-sync` the same way `getRecords()` does. Wonder if I should create a new helper function for `getRecord()`, or what do you reckon?

Reply:
ah, looks like one from `@latticexyz/store` might work

Reply:
worked perfectly, great suggestion